vars/runInspector.groovy

```diff
@@ -28,109 +29,228 @@ import java.util.*
+            println("Agent health: ${agentStatus}")
+            timeout += 1
+            TimeUnit.SECONDS.sleep(5);
+        } else {
+            println("Waited 10 minutes for the agent to become healthy, skipping waiting")
+            agentStatus = 'HEALTHY'
+        }
+    }
+
+    // Query stack for inspector assessment template arn (must be an output)
+    def template_arn = cloudformation(
+            stackName: stackName,
+            queryType: 'output',
+            query: 'TemplateArn',
+            region: body.region
+    )
+
+    // Run the inspector test
+    def assessmentArn = assessmentRun(template_arn)
+    println('Inspector test(s) started')
+
+    // Wait for the inspector test to run
+    def runStatus = getRunStatus(assessmentArn)
+    while  (runStatus != "COMPLETED") {
+          runStatus = getRunStatus(assessmentArn)
+          println("Test Run Status: ${runStatus}")
+          TimeUnit.SECONDS.sleep(5);
+    }
+
+    // This waits for inspector to finish up everything before an actaul result can be returned, this is not waiting for the test to finish
+    def testRunning = true
+    while (testRunning.equals(true)) {
+          def getResults = getResults(assessmentArn).toString()
+          println("Cleanup Status: ${getResults}")
+          if ((getResults.contains("WORK_IN_PROGRESS")).equals(false)) {
+                testRunning = false
+          }
+    }
+
+    // Get the results of the test, write to jenkins and fromated the result to check if the test(s) passed
+    getResults = getResults(assessmentArn)
+    def urlRegex = /http.*[^}]/
+    def resutlUrl = (getResults =~ urlRegex)
+    resutlUrl = resutlUrl[0]
+    def fullResult = resutlUrl.toURL().text
+    writeFile(file: 'Inspector_test_reults.html', text: fullResult)
+    archiveArtifacts(artifacts: 'Inspector_test_reults.html', allowEmptyArchive: true)
+    def findings = formatedResults(assessmentArn)
+    cleanUp(stackName, body.region, bucketName, fileName)
+    return findings
+}
 
-      // Fail the pipeline if insepctor tests did not pass and flag either set to true or not set
-      def failonfinding = body.get('failonfinding', true)
 
-      if (testPassed >= 1){
-          if (!failonfinding) {
-              println("One or more interpector test(s) failed on the AMI however \'failonfinding\' is set to \'False\' and hence the pipeline has not failed")
-              return testPassed
-          }
-          else {
-              throw new GroovyRuntimeException("One or more interpector test(s) failed on the AMI")
-          }
-      }
-      else {
-          return testPassed
-      }
+
+def checkFail(failon, findings){
+    // Make a list of severity levels with >= 1 nFindings
+    def severties = findings.keySet()
+    def severityFindings = []
+    for (s in severties){
+        if (findings[s].toInteger() > 0 ) {
+            severityFindings.add(s)
+        }
+    }
+
+    def validSeverity = ['']
+    switch(failon) {
```

Comment:
lower or upper the case here so you don't have any issues with different input cases on the failon property.

Reply:
This should be fine as I cleanup the input here     
44    ` def failon = body.get('failon', 'Medium').toLowerCase().capitalize()`

Reply:
i would add `.toString()` before `.toLowerCase().capitalize()` in case of a non string input